File: TwoPointers/csharp/IsSubsequence.cs
namespace TwoPointersSolutions;

public static class IsSubsequence
{
    public static bool Check(string target, string source)
    {
        var i = 0;
        var j = 0;
        while (i < target.Length && j < source.Length)
        {
            if (target[i] == source[j])
            {
                i++;
            }
            j++;
        }

        return i == target.Length;
    }
}

File: TwoPointers/java/IsSubsequence.java
public final class IsSubsequence {
  private IsSubsequence() {
  }

  public static boolean check(String target, String source) {
    int i = 0;
    int j = 0;

    while (i < target.length() && j < source.length()) {
      if (target.charAt(i) == source.charAt(j)) {
        i++;
      }
      j++;
    }

    return i == target.length();
  }
}
